pkg/debuginfo/debuginfod.go

```diff
@@ -184,10 +188,7 @@ func (c *HTTPDebuginfodClient) request(ctx context.Context, u url.URL, buildID s
 	// this request will result in a binary object that contains the customary .*debug_* sections.
 	u.Path = path.Join(u.Path, "buildid", buildID, "debuginfo")
 
-	// Uses the default tracer provider and propagator that's set in tracer package,
-	// or from the span context passed in.
-	ctx = httptrace.WithClientTrace(ctx, otelhttptrace.NewClientTrace(ctx))
-
+	ctx = httptrace.WithClientTrace(ctx, otelhttptrace.NewClientTrace(ctx, otelhttptrace.WithTracerProvider(c.tp)))
```

Comment:
While I like this a lot I'm currently re-working this code pretty significantly, can we revert this here, and I promise I'll add this instrumentation! :D

Reply:
Removed those changes 👍